mintapi/api.py

```diff
@@ -237,13 +242,73 @@ def get_transactions_json(self, include_investment=False,
                 expected_content_type='text/json')
             data = json.loads(result.text)
             txns = data['set'][0].get('data', [])
+            df = pd.DataFrame(txns)
+            if start_date:
+                dates = list(df['odate'])
+                cy = datetime.isocalendar(date.today())[0]
+                try:
+                    first_dt = datetime.strptime(dates[0] + str(cy), '%b %d%Y')
+                except:
+                    first_dt = datetime.strptime(dates[0], '%m/%d/%y')
+                if first_dt < start_date:
+                    break
             if not txns:
                 break
             all_txns.extend(txns)
             offset += len(txns)
-
         return all_txns
 
+    def _dateconvert(self, dateraw):
+        # Converts dates from json data
+        cy = datetime.isocalendar(date.today())[0]
+        try:
+            newdate = datetime.strptime(dateraw + str(cy), '%b %d%Y')
+        except:
+            newdate = datetime.strptime(dateraw, '%m/%d/%y')
+        return newdate
+
+    def _debit_credit(self, row):
+        # Reverses credit balances
+        dic = {False: -1, True: 1}
+        return float(row['amount'][1:].replace(',', '')) * dic[row['isDebit']]
+
+    def get_detailed_transactions(self, start_date=None,
+                                  include_investment=False,
+                                  skip_duplicates=False,
+                                  remove_pending=True):
+        """Returns the JSON transaction data as a DataFrame, and converts
+        current year dates and prior year dates into consistent datetime
+        format, and reverses credit activity.
+
+        Note: start_date must be in format mm/dd/yy. If pulls take too long,
+        use a more recent start date. See json explanations of
+        include_investment and skip_duplicates.
+
+        Also note: Mint includes pending transactions, however these sometimes
+        change dates/amounts after the transactions post. They have been
+        removed by default in this pull, but can be included by changing
+        include_pending to False
+
+        """
+        if not pd:
+            raise ImportError(
+                'transactions data requires pandas; '
+                'please pip install pandas'
+            )
```

Comment:
This chunk of code is now duplicated. Would you mind making a function like "assert_pd" that has this code and then can just be called in one line in the places where we use it?